docs/understanding-airbyte/airbyte-protocol-docker.md

```diff
@@ -0,0 +1,49 @@
+# Airbyte Protocol Docker Interface
+
+## Summary
+The [Airbyte Protocol](airbyte-protocol.md) describes a series of structs and interfaces for building data pipelines. The Protocol article describes those interfaces in language agnostic pseudocode, this article transcribes those into docker commands. Airbyte's implementation of the protocol is all done in docker. Thus, this reference is helpful for getting a more concrete look at how the Protocol is used. It can also be used as a reference for interacting with Airbyte's implementation of the Protocol.
+
+
+## Source
+
+### Pseudocode:
+
+```
+spec() -> ConnectorSpecification
+check(Config) -> AirbyteConnectionStatus
+discover(Config) -> AirbyteCatalog
+read(Config, ConfiguredAirbyteCatalog, State) -> Stream<AirbyteRecordMessage | AirbyteStateMessage>
+```
+
+### Docker:
+```shell
+docker run --rm -i <source-image-name> spec
```

Comment:
if this is meant to be conceptual, I'd remove the `--rm -i`. Also, I would add the `-v` since that might confusing